### ISSUE TITLE
error[E0599]: no method named `to_f32` found for struct `euclid::Rect<Au, LayoutPixel>` in the current scope

### ISSUE THREAD

**Issue by kaiwk** [servo/app_units#46]
I want to convert `au_rect: Rect<Au, LayoutPixel>` to `f32_rect: Rect<f32, LayoutPixel>` by `au_rect.to_f32()`, as the compiler prompts: 

```
26 | pub struct Au(pub i32);
   | ----------------------- doesn't satisfy `Au: num_traits::cast::NumCast`
   |
   = note: the method `to_f32` exists but the following trait bounds were not satisfied:
           `Au: num_traits::cast::NumCast`
```

Can we just impl `NumCast` for `Au`?

**Comment by emilio** [comment]
Hmm, do you expect the target rect to be in actual pixels I guess? (as in, the equivalent to `to_f32_px()`?)

I think `NumCast` can be confusing, because you may want the `Au` value as `f32` (so `value.0 as f32` or such), or the actual pixel value as `f32` (`value.as_f32_px()`)

**Comment by kaiwk** [comment]
> as in, the equivalent to to_f32_px()?

Yes, exactly. It seems there is no way to convert the `Rect<Au>` directly.

> because you may want the Au value as f32 (so value.0 as f32 or such)

ha, I see.

**Comment by mrobinson** [comment]
> I think `NumCast` can be confusing, because you may want the `Au` value as `f32` (so `value.0 as f32` or such), or the actual pixel value as `f32` (`value.as_f32_px()`)

Out of curiosity, when is it useful to have `value.0 as f32`?
